internal/provider/waypoint/resource_waypoint_add_on_definition.go

```diff
@@ -0,0 +1,563 @@
+				Description: "List of labels attached to this Add-on Definition.",
+				ElementType: types.StringType,
+				PlanModifiers: []planmodifier.List{
+					listplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"readme_markdown_template": schema.StringAttribute{
+				Description: "The markdown template for the Add-on Definition README.",
+				Optional:    true,
+			},
+			"terraform_cloud_workspace_details": &schema.SingleNestedAttribute{
+				Required:    true,
+				Description: "Terraform Cloud Workspace details",
+				Attributes: map[string]schema.Attribute{
+					"name": &schema.StringAttribute{
+						Required:    true,
+						Description: "Name of the Terraform Cloud Workspace",
+					},
+					"terraform_project_id": &schema.StringAttribute{
+						Required:    true,
+						Description: "Tetraform Cloud Project ID",
+					},
+				},
+			},
+			"terraform_no_code_module": &schema.SingleNestedAttribute{
+				Required:    true,
+				Description: "Terraform Cloud No Code Module details",
+				Attributes: map[string]schema.Attribute{
+					"source": &schema.StringAttribute{
+						Required:    true,
+						Description: "No Code Module Source",
+					},
+					"version": &schema.StringAttribute{
+						Required:    true,
+						Description: "No Code Module Version",
+					},
+				},
+			},
+		},
+	}
+}
+
+func (r *AddOnDefinitionResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*clients.Client)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *http.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+func (r *AddOnDefinitionResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan *AddOnDefinitionResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	projectID := r.client.Config.ProjectID
+	if !plan.ProjectID.IsUnknown() {
+		projectID = plan.ProjectID.ValueString()
+	}
+
+	orgID := r.client.Config.OrganizationID
+	loc := &sharedmodels.HashicorpCloudLocationLocation{
+		OrganizationID: orgID,
+		ProjectID:      projectID,
+	}
+
+	client := r.client
+	ns, err := getNamespaceByLocation(ctx, client, loc)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"error getting namespace by location",
+			err.Error(),
+		)
+		return
+	}
+
+	stringLabels := []string{}
+	if !plan.Labels.IsNull() && !plan.Labels.IsUnknown() {
+		diagnostics := plan.Labels.ElementsAs(ctx, &stringLabels, false)
+		if diagnostics.HasError() {
+			resp.Diagnostics.AddError(
+				"error converting labels",
+				"Failed to convert labels from types.List to string list",
```

Comment:
nit: this error likely bubbles up to the end user, in which case `from types.List to string list` is likely too low level or implementation specific details that we don't need to share. A more generic error about the list formatting might be more appropriate